### ISSUE TITLE
Variable class should not be an error

### ISSUE THREAD

**Issue by metal3d** [microsoft/pyright#1628]
Let's make a conditional class definition:

```python
class Foo:
   pass

class Bar:
   pass

# The "Base" class can be changed, for example for some platforms
Base = Foo
if os.platform == "windows":
   Base = Bar

class Baz(Base):
   pass
```

Pyright says:
`reportGeneralTypeIssue [E] Argument to class should be a base class`

It's not a problem in Python execution, but pyright gives an error.

A real example, I want to use "ThemedTk" in Linux, not on others platforms:
```python
from tkinter import tk

THEMED_OS = ["linux"]

TK = tk.Tk
if sys.platform in THEMED_OS:
    from ttkthemes import ThemedTk as TK

class MyGui(TK):
    """ The Graphical User Interface """
   def __init__(self):
      super().__init__()
    
      # only provided by ThemedTk
      set_theme = getattr(self, "set_theme")
      if set_theme:
          set_theme("arc")
```

**Comment by erictraut** [comment]
This code is too dynamic for a static type checker.

What you can do is this:
```python
import tkinter as tk
import sys

if sys.platform == "linux":
    from ttkthemes import ThemedTk as TK
else:
    TK = tk.Tk
```

**Comment by metal3d** [comment]
Hi.

It says that code is unreachable. So pyright should accept one or the other.

**Comment by metal3d** [comment]
Sorry but, retried and it fails a lot.
Take this example (after several attempts, with `try` or `if` somthing):

```python
# Fail...

try:
    from ttkthemes import ThemedTk as TK
except ImportError:
    TK = tk.Tk

class App(TK):
   """Class..."""
```

```python
# Fail also

foundttk = False
if importlib.find_loader("ttkthemes"):
    print("Found")
    foundttk = True

if foundttk:
    from ttkthemes import ThemedTk

    TK = ThemedTk
else:
    TK = tk.Tk

class App(TK):
    """Simple Application"""

```

Only this is OK:

```python

if True:
    from ttkthemes import ThemedTk

    TK = ThemedTk
else:
    TK = tk.Tk

class App(TK):
    """Simple Application"""

```

There is no reason that `if sys.platform == "linux"` works while `if foundttk` fails.

**Comment by erictraut** [comment]
Static type checkers don't "execute" your code. They generally assume that all code paths are potentially executed at runtime.

[PEP 484](https://www.python.org/dev/peps/pep-0484/#id35) specifically lists a few patterns like `if sys.platform == "linux"` and `if True` that type checkers are expected to evaluate statically to determine whether particular blocks are executed. If you want your code to work with static type checkers, you need to stick to these simple checks.
